wordcount/src/sort/Main.java

```diff
@@ -0,0 +1,77 @@
+        String newString = rawString.replaceAll("[[\\.\\?\\!\\,\\;\\:\\{\\}\\(\\)\\']]", ""); //this is the parsed string
+         
+        String[] words = newString.split(" +"); //split and store into words[]
+
+        HashMap<String, Integer> wordHashMap = new HashMap<String, Integer>();
+
+        for (String s : words)
+        {
+            Integer f = wordHashMap.get(s.toLowerCase());
+
+            if (f == null)
+            {
+                wordHashMap.put(s.toLowerCase(), 1);
+            }
+            else 
+            {
+                
+                wordHashMap.put(s.toLowerCase(), f += 1);
+            }
+
+        }
+
+        System.out.println(" *** Top 50 Sorted by quantity ***");
+        
+        ArrayList<HashMap.Entry<String, Integer>> sortedList = new ArrayList<HashMap.Entry<String, Integer>>();
+        sortedList.addAll(wordHashMap.entrySet());
+
+        Collections.sort(sortedList, new Comparator<Map.Entry<String, Integer> >()
+        {
+            public int compare (HashMap.Entry<String, Integer> o1, HashMap.Entry<String, Integer> o2)
+            {
+                return o2.getValue() - o1.getValue();
+            }
+        });
+
+        ArrayList<HashMap.Entry<String, Integer>> topList = new ArrayList<HashMap.Entry<String, Integer>>();
+
+        for(int i = 0; i <= 50; i++)
+        {
+            topList.add(sortedList.get(i));
+            System.out.println(sortedList.get(i));
+        }
+        
+
+        
+        System.out.println(" *** Top 50 Sorted by Alphabetically ***");
+        
+        Collections.sort(topList, new Comparator<Map.Entry<String, Integer> >()
+        {
+            public int compare (HashMap.Entry<String, Integer> o1, HashMap.Entry<String, Integer> o2)
+            {
+                return o1.getKey().compareToIgnoreCase(o2.getKey());
+            }
+        });
+        
+        for(int i = 0; i <= 50; i++)
+        {
+            System.out.println(topList.get(i));
+        }
```

Comment:
In loops, when starting from `0` and ending on the last element of the array, use `<` instead of `<=` - this will ensure that you don't end up having an extra array elements when you're done with your loop **(this disntinction is also highly important in avoiding errors - unlike Javascript, going past the full length of an array will generate an error `ArrayIndexOutOfBounds`)**.